### ISSUE TITLE
Release a new version

### ISSUE THREAD

**Issue by ryandesign** [LLNL/yorick#39]
yorick 2.2.04 doesn't build with llvm.org clang 18:

```
graph.c:4098:17: error: parameter 'argc' was not declared, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
 4098 | Y_current_mouse(argc)
      |                 ^
 4099 | {
graph.c:4098:1: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
 4098 | Y_current_mouse(argc)
      | ^
```

This bug was fixed in 2018 in 83d52189401e8a07a65203a6f2edafef52fb133e but there has not been a new release since then.

Please release a new version so that [everyone who packages your software](https://repology.org/project/yorick/versions) does not have to independently research how to fix this issue and other issues that you've already fixed in the nine years since your last release.

**Comment by dhmunro** [comment]
Yorick was abandoned nearly ten years ago.  There have been a few bug fixes for legacy users at LLNL and for yeti since then, but 2.2.04 is the final version.  You should switch to python/numpy/scipy as soon as possible; certainly it is a serious mistake to write new yorick interpreted code, as inevitably yorick will eventually fail to build on more and more systems.  As a stopgap measure, you can use the LLNL/pyorick repo to incrementally migrate to python.

That said, I don't really understand the problem you are having.  What do you mean by "release"?  Do you simply want a new version number?  If so, why?  Isn't your packaging system set to grab the latest source from this LLNL/yorick repo, or are you building from the old y_2_2_04 tag instead of from the latest commit?  Why are you not getting the latest "2.2.04" in the first place?  If the problem is that you are getting your source from somewhere other than this LLNL/yorick repo, there isn't much I can do about that.  I don't know that there is anyone to update the packages for various distros (e.g.- Debian), even if the version number were bumped to, say, 2.2.05 - I believe there is a bunch of manual work by the "maintainer" of the package for each distro that would still have to be done even if the version number here changes.  Since that's the case, it would be better if those packages separately added a new subversion number if that is required for some particular case.

In any event, I don't believe that incrementing the version number of this repo will by itself fix whatever problem you are having.  Yorick died at age 2.2.04, and I really have no desire to change the date on its gravestone.

**Comment by spytheman** [comment]
Why did it die?

**Comment by spytheman** [comment]
Were there technical reasons for that, or just lack of enough interest?

**Comment by dhmunro** [comment]
I wrote yorick in 1991-2 and LLNL released it as open source software in 1993.  Its purpose was to provide a flexible way to analyze and interpret the results of numerical simulations of laser fusion targets.  Yorick was always a one man show - it turns out that merely releasing software (on sourceforge.net at the time) does not guarantee community interest.  The MIT team that developed numpy (or its precursor - I forget what it was called) did use yorick as a performance benchmark, tweaking their code until it ran at roughly the same speed (about 5 times slower than compiled code for typical vector operations and a few hundred times slower for scalar code - that is, code with indexing operations and loops).  Yorick was similarly performance normalized to the commercial IDL language (which became PVWave and other products when it was eventually sold).

By around 2010 it was becoming obvious even to me that no sane developer (let alone physicist like me) could ever choose yorick with its one person support (total - maybe three of us have contributed lines of code to what's in this repo), when with scipy they could get a package that did everything yorick could ever do and much more, and had hundreds of people supporting it.  And despite the fact that ndarrays are second class python objects, python as a programming language is a far, far stronger language than yorick ever was, with every bit of it critiqued in detail by a very large number of very talented software engineers.  By 2012 or 2013, I began to actively push people away from yorick to scipy, and myself abandoned using yorick for my work as a physicist.  I believe that all the commercial alternatives to scipy - say Matlab or Mathematica - have the same problem keeping up with scipy: They simply don't have a support staff that can keep pace with the hundreds of scipy contributors.  Even if they do manage to invent some new feature, it's doomed to be rapidly equaled then surpassed by scipy enthusiasts, and none has an underlying language remotely as strong as python.

In hindsight, the lack of interest that doomed yorick happened circa 1995.  To be successful, an open source project needs people to become interested early.  Part of this is hard work - van Rossum was working very hard to generate enthusiasm for python at that time - but part is just luck.  I believe the underlying truth is that it really doesn't matter which choice is made; it's more important that a bunch of people make the same decision and begin building together.  I would have been far better off abandoning yorick in 1995, and committing myself to numpy, where I might have had more impact.  However, I have no serious regrets; except for 1991-2, yorick was never the largest part of my job as a physicist working on laser fusion.

By now, there are also many technical reasons not to use yorick.  It was written to compile and run on machines as they were thirty years ago.  For example, yorick interactive graphics - one of the main reasons anyone used it - are deeply rooted in the dangerously obsolete X11 windowing system.  Modern GPUs simply do not work with the primitives X11 assumed.  Updating yorick graphics to modern standards needed to guarantee it will continue to work would be a major undertaking - many man-months of dedicated effort at least.  I did a pretty decent job of making yorick portable, but thirty years is a long time.

The bottom line is no one should begin using  or studying yorick today.  Learn scipy instead.
